numba/targets/arraymath.py

```diff
@@ -2396,6 +2396,21 @@ def np_corrcoef_impl_single_variable(x, y=None, rowvar=True):
 #----------------------------------------------------------------------------
 # Element-wise computations
 
+@overload(np.flatnonzero)
+def np_flatnonzero(a):
+
+    if numpy_version < (1, 15):
+        if not isinstance(a, types.Array):
+            raise TypingError("Argument 'a' must be an array")
+            # numpy raises an Attribute error with:
+            # 'xxx' object has no attribute 'ravel'
+
+    def impl(a):
```

Comment:
I think that this function needs guarding with the **new** `type_can_asarray` helper:

https://github.com/numba/numba/blob/8bfe1e845d09b7ecc2055a2be4f90a066145bab6/numba/numpy_support.py#L599-L604

so e.g. someone randomly passing a string to this function doesn't get a weird error message from `np.asarray` about whatever it doesn't like about a string.

Reply:
Good call.

I added a couple of lines to handle invalid array-likes; all seems fine except for the python 2.7 builds, which error with:  ```TypeError: can't unbox str type```

Let me know what the best course of action is and I will tweak accordingly.

Reply:
Thanks. String support isn't present in Numba for Python 2.7, which is probably the root cause, think it's fine to add a guard around Python version in the test inputs for this specific case.